src/stores/CommunityPrototypeStore.ts

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { AsyncStoreWithClient } from "./AsyncStoreWithClient";
+import defaultDispatcher from "../dispatcher/dispatcher";
+import { ActionPayload } from "../dispatcher/payloads";
+import { Room } from "matrix-js-sdk/src/models/room";
+import { EffectiveMembership, getEffectiveMembership } from "../utils/membership";
+import SettingsStore from "../settings/SettingsStore";
+import * as utils from "matrix-js-sdk/src/utils";
+import { UPDATE_EVENT } from "./AsyncStore";
+
+interface IState {
+    // nothing of value - we use account data
+}
+
+export interface IRoomProfile {
+    displayName: string;
+    avatarMxc: string;
+}
+
+export class CommunityPrototypeStore extends AsyncStoreWithClient<IState> {
+    private static internalInstance = new CommunityPrototypeStore();
+
+    private constructor() {
+        super(defaultDispatcher, {});
+    }
+
+    public static get instance(): CommunityPrototypeStore {
+        return CommunityPrototypeStore.internalInstance;
+    }
+
+    protected async onAction(payload: ActionPayload): Promise<any> {
+        if (!this.matrixClient || !SettingsStore.getValue("feature_communities_v2_prototypes")) {
+            return;
+        }
+
+        if (payload.action === "MatrixActions.Room.myMembership") {
+            const room: Room = payload.room;
+            const membership = getEffectiveMembership(payload.membership);
+            const oldMembership = getEffectiveMembership(payload.oldMembership);
+            if (membership === oldMembership) return;
+
+            if (membership === EffectiveMembership.Invite) {
+                try {
+                    const path = utils.encodeUri("/rooms/$roomId/group_info", {$roomId: room.roomId});
+                    const profile = await this.matrixClient._http.authedRequest(
+                        undefined, "GET", path,
+                        undefined, undefined,
+                        {prefix: "/_matrix/client/unstable/im.vector.custom"});
```

Comment:
No particular need to change anything now, but just pointing out that `im.vector` should probably be replaced with `io.element` for new work.

Reply:
👍 will keep in mind for the next one (which should hopefully not need a generic `custom` namespace and instead can be associated to an MSC)